File: clients/csharp/Model/Vec2Double.cs
namespace AiCup2019.Model
{
    public struct Vec2Double
    {
        public double X { get; set; }
        public double Y { get; set; }
        public Vec2Double(double x, double y)
        {
            this.X = x;
            this.Y = y;
        }
        public static Vec2Double ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new Vec2Double();
            result.X = reader.ReadDouble();
            result.Y = reader.ReadDouble();
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(X);
            writer.Write(Y);
        }
    }
}

File: clients/fsharp/Model/Vec2Double.fs
#nowarn "0058"
namespace AiCup2019.Model
type Vec2Double = {
    X: double;
    Y: double;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write this.X
        writer.Write this.Y
    static member readFrom(reader: System.IO.BinaryReader) = {
        X = reader.ReadDouble()
        Y = reader.ReadDouble()
    }
